codalab/model/bundle_model.py

```diff
@@ -981,6 +981,7 @@ def delete_bundles(self, uuids):
             connection.execute(
                 cl_bundle_dependency.delete().where(cl_bundle_dependency.c.child_uuid.in_(uuids))
             )
+            connection.execute(cl_worker_run.delete().where(cl_worker_run.c.run_uuid.in_(uuids)))
```

Comment:
Hm, this shouldn't be necessary because you can only delete bundles that aren't running...if this is needed, that means there must be a deeper problem somewhere...

If you put this here, you should put a note to say "in case something goes wrong".

Reply:
If this is the case, we probably should change the definition of `worker_run` table, as it referenced `worker_run.run_uuid` as a foreign key. Check this out: https://github.com/codalab/codalab-worksheets/blob/master/codalab/model/tables.py#L336

Reply:
Ignore my previous comment, foreign key constraints should be there. I'll add the comment.

Reply:
I double-checked those records that were causing the error in `worker_run` table, they actually don't exist at all, which means our code logic is actually correct. But since we referenced `worker_run.run_uuid` as the foreign key to `bundle.uuid`, MySQL throws an error.

Reply:
Only 2 out of 512 bundles do exist in `worker_run` table:
```
+---------+--------------------------+------------------------------------+
| user_id | worker_id                | run_uuid                           |
+---------+--------------------------+------------------------------------+
| 0       | vm-clws-prod-gpuworker-5 | 0xd905a9aff8674e1385fa74ceef660bbf |
| 0       | vm-clws-prod-gpuworker-2 | 0x2948f06416e446ebba3b517e109f8aee |
+---------+--------------------------+------------------------------------+
2 rows in set (0.01 sec)
```